### PR TITLE
OCPBUGS-37711: fix: use marker annotation to determine successful wipe and rewipe on failure

### PR DESCRIPTION
**Pull request by jakobmoellerdev**

### CODE REVIEW

File: internal/controllers/vgmanager/wipe_devices.go [hunk 1]
```diff
@@ -4,62 +4,96 @@ import (
...
+		} else if deviceWiped {
+			updated = true
 		}
-		if deviceWiped {
-			wiped = true
+	}
+
+	if updated {
+		if volumeGroup.Annotations[constants.DevicesWipedAnnotation] == "" {
+			volumeGroup.Annotations = make(map[string]string)
```
Comment (suleymanakbas91) [inline internal/controllers/vgmanager/wipe_devices.go:47]: Are we overriding all the annotations here?

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]

File: internal/controllers/vgmanager/wipe_devices.go (hunk 1 above)
Comment (jakobmoellerdev) [reply internal/controllers/vgmanager/wipe_devices.go:47]: I had a brain lag and didnt compare for map==nil but for the key, that should be fixed now

File: internal/controllers/vgmanager/wipe_devices.go [hunk 2]
```diff
@@ -4,62 +4,96 @@ import (
...
 			logger.Info(fmt.Sprintf("skipping wiping optional device %s: %v", path, err))
+		} else if deviceWiped {
+			updated = true
 		}
-		if deviceWiped {
-			wiped = true
+	}
+
+	if updated {
+		if volumeGroup.Annotations[constants.DevicesWipedAnnotation] == "" {
```
[the remark this answers is not in the document]
Comment (jakobmoellerdev) [reply internal/controllers/vgmanager/wipe_devices.go]: see above, thanks for calling out

**Pull request merged**